.test-defs/cmd/create-shoot/helper.go

```diff
@@ -0,0 +1,146 @@
+package main
+
+import (
+	"fmt"
+	"sort"
+
+	gardenv1beta1 "github.com/gardener/gardener/pkg/apis/garden/v1beta1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// getSeed returns the first seed ordered by creation date with the provided cloudprofile.
+func getSeed(gardenClusterClient kubernetes.Interface, cloudprofileName string) (*gardenv1beta1.Seed, error) {
+
+	seedObjects, err := gardenClusterClient.Garden().Garden().Seeds().List(metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	seeds := []*gardenv1beta1.Seed{}
+	for _, seedObject := range seedObjects.Items {
+		seed := seedObject
+		if seed.Spec.Cloud.Profile == cloudprofileName && *seed.Spec.Visible == true && *seed.Spec.Protected == false {
+			seeds = append(seeds, &seed)
+		}
+	}
+
+	if len(seeds) == 0 {
+		return nil, fmt.Errorf("Cannot find a seed for cloudprofile %s", cloudprofileName)
+	}
+
+	sort.Slice(seeds, func(i, j int) bool {
+		return seeds[i].ObjectMeta.CreationTimestamp.Before(&seeds[j].ObjectMeta.CreationTimestamp)
+	})
+
+	return seeds[0], nil
+}
+
+// getDefaultDomain returns the default domain of a garden cluster by searching the garden namespace for the secret with the label "garden.sapcloud.io/role=default-domain".
+func getDefaultDomain(gardenClusterClient kubernetes.Interface) (string, error) {
+	secrets, err := gardenClusterClient.ListSecrets("garden", metav1.ListOptions{LabelSelector: "garden.sapcloud.io/role=default-domain"})
+	if err != nil {
+		return "", err
+	}
+	if len(secrets.Items) == 0 {
+		return "", fmt.Errorf("Cannot find a default domain")
+	}
+	return secrets.Items[0].Annotations["dns.garden.sapcloud.io/domain"], nil
+}
+
+// getSecretBindingName returns the name of the secretbinding that corresponds to a cloudprofile in a project namespace.
+func getSecretBindingName(gardenClient kubernetes.Interface, projectNamespace, cloudprofileName string) (string, error) {
```

Comment:
`getSecretBindingName ` and `getCloudprofile` - why is this needed in the first place? Can't we just get these variables from the ENV (like some other inputs)?

> `LabelSelector: fmt.Sprintf("cloudprofile.garden.sapcloud.io/name=%s", cloudprofileName)`

This label is no contract/not part of any API, so you can't rely on it - it's just used by the dashboard.

Reply:
Is there another way to get any parameters programmatically (like region) or is it better to get all needed parameters with env vars?
We want the input parameters to be as little as possible.

Reply:
I'd recommend to get these things (cloudprofile, region, secret) as inputs (just like it was with the old integration test).
For cloudprofile/region you could theoretically list all seeds and get one that matches to the desired cloud, however, you would not be able to find a proper secretbinding programmatically (this is not coupled in any way to the cloud provider - as said, the label is just for the dashboard).
--> Better get everything as inputs.

Reply:
Done